modules/light-clients/07-tendermint/types/store.go

```diff
@@ -1,18 +1,25 @@
 package types
 
 import (
+	"encoding/binary"
 	"strings"
 
 	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
 	host "github.com/cosmos/ibc-go/modules/core/24-host"
 	"github.com/cosmos/ibc-go/modules/core/exported"
 )
 
-// KeyProcessedTime is appended to consensus state key to store the processed time
-var KeyProcessedTime = []byte("/processedTime")
+const KeyIterateConsensusStatePrefix = "iterateConsensusStates"
+
+var (
+	// KeyProcessedTime is appended to consensus state key to store the processed time
+	KeyProcessedTime = []byte("/processedTime")
+	KeyIteration     = []byte("/iterationKey")
```

Comment:
Could you add a very long detailed comments at the top of this file explaining why we have `KeyIteration`, what purpose it serves and that (IMO) it should eventually be replaced by changing how the keys are stored (when we update to a breaking version of IBC)

I think we should also open an issue to track removing this workaround as I don't see this fix as a sustainable, but very useful in the short term

Reply:
Sure, but i imagine this might just be something that sticks around. It's not very costly to do this (from a state space or developer maintanence point of view), and its a pain to change from a operational point of view.

It may get packaged with other breaking changes in IBC 2.0